### ISSUE TITLE
Project is missing a Code of Coduct

### ISSUE THREAD

**Issue by dvberkel** [gnab/remark#600]
A [code of conduct](https://en.wikipedia.org/wiki/Code_of_conduct)

> A code of conduct is a set of rules outlining the norms, rules, and responsibilities of, and or proper practices for, an individual.

Currently this project is missing one. This could alienate a lot of people. E.g. the [comment](https://github.com/gnab/remark/issues/553#issue-394036139) on issue #553 could isn't constructive and can zap the energies of maintainers.

A code of conduct could help in delineate what is acceptable behavior and provide directions how to deal with persons that violate the code of conduct.

**Comment by abelards** [comment]
I'm all for a code of conduct but I'll let @gnab choose.

If we go for a CoC, I suggest the Contributor Covenant:
https://www.contributor-covenant.org/version/2/0/code_of_conduct

Adding a CoC is easy, enforcing it consistently is hard.

I can see that the comment you mention is out of the "standard" that is suggested, but not really in the "unacceptable behaviours". My candid question is: if we decide that it is off limits, what would you recommend we do? Close/erase the issue?

Thanks for mentioning the topic, I'm looking forward to the contributors' opinions.

**Comment by dvberkel** [comment]
> Adding a CoC is easy, enforcing it consistently is hard.
>

I totally agree that enforcing the code of conduct can be hard. I do think that it is worth the effort.
 
> I can see that the comment you mention is out of the "standard" that is suggested, but not really in the "unacceptable behaviours". My candid question is: if we decide that it is off limits, what would you recommend we do? Close/erase the issue?

I think I would address the original poster of the issue that their comment are at least of topic and that it could read as not constructive. The maintainers are more than willing to contribute to the project, including responding to issues, but that takes a lot of energy. Responding to non-constructive issues can zap that energy and the motivation to work on the project. So please be considerate of your words and be constructive.

Depending on the reaction of the original poster, we could take more definite actions.

**Comment by abelards** [comment]
Agreed on both points. I did not mean "enforcing is hard" to antagonize or "vote against", I was simply asking what you had in mind. As a new contributor I'm still keeping a low profile when suggesting fixes or closing tickets, but I'll perhaps add some comments about the tone soon.

**Comment by peterj** [comment]
I think we should add CoC as well. It is simple enough through Github to just add it. By default, we can add a Code of Conduct "that is adapted from the Contributor Covenant, version 1.4, available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html". 

I am not familiar with the exact adaptations that were made or different versions though. 

One thing that is required to add is an email address where the project team can be contacted at - not sure what the practice here is (i.e. create a Gmail account that maintainers have access to?)

**Comment by gnab** [comment]
To me this boils down to common courtesy, or the lack of such, which has never really been a problem from what I can recall. But if you guys think a CoC is something we should have, I'm all for it! 👍
